src/apm/health.go

```diff
@@ -0,0 +1,185 @@
+
+	var healthMountPath string
+	{
+		v, _ := getValueFromEnv(container.Env, envHealthFleetControlFilepath)
+		if healthMountPath, err = i.validateHealthFilepath(v); err != nil {
+			return *originalPod, fmt.Errorf("invalid env value %q for %q > %w", v, envHealthFleetControlFilepath, err)
+		}
+	}
+
+	// set defaults
+	for _, entry := range healthDefaultEnv {
+		if idx := getIndexOfEnv(sidecarContainerEnv, entry.Name); idx == -1 {
+			sidecarContainerEnv = append(sidecarContainerEnv, corev1.EnvVar{
+				Name:  entry.Name,
+				Value: entry.Value,
+			})
+		}
+	}
+
+	sidecarListenPort := defaultHealthListenPort
+	if v, ok := getValueFromEnv(sidecarContainerEnv, envHealthListenPort); ok {
+		sidecarListenPort, err = i.validateHealthListenPort(v)
+		if err != nil {
+			return *originalPod, fmt.Errorf("invalid env value %q for %q > %w", v, envHealthListenPort, err)
+		}
+	}
+
+	if isContainerVolumeMissing(&pod.Spec.Containers[firstContainer], healthVolumeName) {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      healthVolumeName,
+			MountPath: healthMountPath,
+		})
+	}
+
+	// We just inject Volumes and init containers for the first processed container.
+	if isInitContainerMissing(pod, healthSidecarContainerName) {
+		if isPodVolumeMissing(pod, healthVolumeName) {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: healthVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				}})
+		}
+
+		restartAlways := corev1.ContainerRestartPolicyAlways
+		sidecarContainer := corev1.Container{
+			Name:          healthSidecarContainerName,
+			Image:         inst.Spec.Agent.Image,
+			RestartPolicy: &restartAlways,
+			VolumeMounts: []corev1.VolumeMount{{
+				Name:      healthVolumeName,
+				MountPath: healthMountPath,
+			}},
+			Ports: []corev1.ContainerPort{
+				{ContainerPort: int32(sidecarListenPort)},
+			},
+			Env: sidecarContainerEnv,
+		}
+
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, sidecarContainer)
+	}
+
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+	pod.Annotations[HealthInstrumentedAnnotation] = "true"
+
+	return pod, nil
+}
+
+func (i *baseInjector) injectEnvVarsIntoTargetedEnvVars(instEnvVars []corev1.EnvVar, containerEnvVars *[]corev1.EnvVar) {
+	for _, env := range instEnvVars {
+		if env.Name == envHealthFleetControlFilepath {
+			if idx := getIndexOfEnv(*containerEnvVars, env.Name); idx == -1 {
+				*containerEnvVars = append(*containerEnvVars, env)
+			}
+			break
+		}
+	}
+}
+
+func (i *baseInjector) injectEnvVarsIntoSidecarEnvVars(instEnvVars []corev1.EnvVar, sidecarEnvVars *[]corev1.EnvVar) {
+	for _, env := range instEnvVars {
+		// configure sidecar specific env vars
+		if idx := getIndexOfEnv(*sidecarEnvVars, env.Name); idx == -1 {
+			*sidecarEnvVars = append(*sidecarEnvVars, env)
+		}
+	}
```

Comment:
Same comment here - 
Recommendation of Go

When to pass the address of a slice:
 1. Modifying the slice header directly:
If you need to change the length or capacity of the slice itself, you'll need to pass the address of the slice header.
2. Avoiding unnecessary copies:
For very large slices, passing the address can be slightly more efficient as it avoids copying the slice header.